pkg/status/combinedstatus.go

```diff
@@ -68,46 +68,42 @@ func (c *Controller) syncCombinedStatus(ctx context.Context, ref string) error {
 		return nil
 	}
 
-	if err = c.updateOrCreateCombinedStatus(ctx, combinedStatus, generatedCombinedStatus); err != nil {
+	generatedCombinedStatus.ResourceVersion = combinedStatus.ResourceVersion // in case of update
+	if err = c.updateOrCreateCombinedStatus(ctx, generatedCombinedStatus); err != nil {
 		return fmt.Errorf("failed to update or create CombinedStatus: %w", err)
 	}
 
 	logger.Info("Synced CombinedStatus", "ns", ns, "name", name)
 	return nil
 }
 
-func (c *Controller) updateOrCreateCombinedStatus(ctx context.Context, combinedStatus,
+func (c *Controller) updateOrCreateCombinedStatus(ctx context.Context,
 	generatedCombinedStatus *v1alpha1.CombinedStatus) error {
-	// set labels
-	combinedStatus.Labels = generatedCombinedStatus.Labels
-	// set results
-	combinedStatus.Results = generatedCombinedStatus.Results
-
 	logger := klog.FromContext(ctx)
-	csEcho, err := c.wdsKsClient.ControlV1alpha1().CombinedStatuses(combinedStatus.Namespace).Update(ctx,
-		combinedStatus, metav1.UpdateOptions{FieldManager: controllerName})
+	csEcho, err := c.wdsKsClient.ControlV1alpha1().CombinedStatuses(generatedCombinedStatus.Namespace).Update(ctx,
+		generatedCombinedStatus, metav1.UpdateOptions{FieldManager: controllerName})
 
 	if err != nil {
 		if errors.IsNotFound(err) {
```

Comment:
BTW, you expect `Update` to succeed if and only if the object's `ResourceVersion` is set to something, and you expect `Create` to succeed if and only if the object's `ResourceVersion` is the empty string, right? This suggests that it is certain which API call to make.